core/src/kmx/kmx_plus.cpp

```diff
@@ -953,6 +953,89 @@ COMP_KMXPLUS_LIST_Helper::getIndex(KMX_DWORD i) const {
   return indices + i;
 }
 
+
+// USET
+
+bool
+COMP_KMXPLUS_USET::valid(KMX_DWORD _kmn_unused(length)) const {
+  if (header.size < sizeof(*this)
+      + (usetCount  * sizeof(COMP_KMXPLUS_USET_USET))
+      + (rangeCount * sizeof(COMP_KMXPLUS_USET_RANGE))) {
+    DebugLog("header.size < expected size");
+    assert(false);
+    return false;
+  }
+  return true;
+}
+
+COMP_KMXPLUS_USET_Helper::COMP_KMXPLUS_USET_Helper() : uset(nullptr), is_valid(false), usets(nullptr), ranges(nullptr) {
+}
+
+bool
+COMP_KMXPLUS_USET_Helper::setUset(const COMP_KMXPLUS_USET *newUset) {
+  DebugLog("validating newUset=%p", newUset);
+  is_valid = true;
+  if (newUset == nullptr) {
+    // null = invalid
+    is_valid = false;
+    // No assert here: just a missing layer
+    return false;
+  }
+  uset = newUset;
+  const uint8_t *rawdata = reinterpret_cast<const uint8_t *>(newUset);
+  rawdata += LDML_LENGTH_USET;  // skip past non-dynamic portion
+  // usets
+  if (uset->usetCount > 0) {
+    usets = reinterpret_cast<const COMP_KMXPLUS_USET_USET *>(rawdata);
+  } else {
+    usets = nullptr;
+    // not invalid, just empty.
+  }
+  rawdata += sizeof(COMP_KMXPLUS_USET_USET) * uset->usetCount;
+  // entries
+  if (uset->rangeCount > 0) {
+    ranges = reinterpret_cast<const COMP_KMXPLUS_USET_RANGE *>(rawdata);
+  } else {
+    ranges = nullptr;
+  }
+
+  // Now, validate offsets by walking
+  if (is_valid) {
```

Comment:
This can never be false

Reply:
this is true.  I guess if there were ranges-but-no-usets it could be invalid. 

I kept this in so that it's parallel code structure to the other helper functions.  it's a dead test though ,so i can remove.